backend/routes/referralRouters.ts

```diff
@@ -15,6 +16,53 @@ referralRouter.get(
   async (req: Request, res: Response, next: NextFunction) => {}
 );
 
+referralRouter.get(
+  "/user",
+  async (req: Request, res:Response, next: NextFunction) => {
+    try{
+      const userId = req.query.userId ? (req.query.userId as string) : "";
+      if(req.user?.id!==userId){ //we should only get referrals if the user logged is the user passed into get request
+        throw new StatusCodedError("unauthorized request: userId does not match", 401);
+      }
+      const referrals = await referralControllers.getReferralsFromUserId(userId);
+      res.status(200).json({referrals});
+    }catch(e:any){
+      next(new Error(e));
+    }
+  }
+);
+
+//=============middleware for getting referrals from a job post
+const checkUserIsManager = (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  if (req.user && req.user.isManager) {
+    next();
+  } else {
+    try {
+      throw new StatusCodedError("unauthroized request: not a manager", 401);
+    } catch (err) {
+      next(err);
+    }
```

Comment:
i think this is equivalent to `next(new StatusCodedError("...", 401))`, which would more clearly communicate the intent. Also since the user is authenticated but forbidden from a particular resource, i think 403 is more appropriate